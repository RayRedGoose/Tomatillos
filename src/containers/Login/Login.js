import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { postSignIn } from '../../apiCalls/apiCalls';
import { addUser } from '../../actions';
import './Login.scss';

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      errorMsg: ''
    }
  }

  handleChange = (event) => {
    let { target } = event;
    if (this.state.email && this.state.password) {
      this.setState({errorMsg: ''});
    }
    switch(target.id) {
      case 'email':
        this.setState({email: target.value})
        break;
      case 'password':
        this.setState({password: target.value})
        break;
      default:
        console.log('hits default');
    }
  }

  validateForm = () => {
    if (this.state.email && this.state.password) {
      this.setState({errorMsg: ''});
      return true;
    } else {
      this.setState({errorMsg: 'Please enter an email and a password'});
      return false;
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    if (this.validateForm()) {
      let user = {
        email: this.state.email,
        password: this.state.password
      };
      this.fetchUser(user);
    }
  }

  fetchUser = async (user) => {
    try {
      const result = await postSignIn(user);
      this.props.addUser(result.user);
      this.props.history.push('/ratings');
      this.setState({
        email: '',
        password: ''
      });
    }
    catch (error) {
      this.setState({errorMsg: 'Your email or password was incorrect'});
    }
  }

  render() {
    return (
      <form className="login-form">
        <h3>Please sign in to continue</h3>
        <div className="input-wrapper">
          <div className="error">{this.state.errorMsg}</div>
          <label>Email:</label>
          <input autoComplete="off" id="email" value={this.state.email} type="text" onChange={(event) => this.handleChange(event)}/>
          <label>Password:</label>
          <input id="password" value={this.state.password} type="password" onChange={(event) => this.handleChange(event)}/>
          <button className="sign-in" onClick={(event) => this.handleSubmit(event)}>Sign in</button>
          <Link id="back-home" to="/">Go back</Link>
        </div>
      </form>
    )
  }
}

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    addUser
  }, dispatch)
)

export default withRouter(connect(null, mapDispatchToProps)(Login));
