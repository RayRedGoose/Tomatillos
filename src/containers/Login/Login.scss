@import '../../variables.scss';

.login-form {
  background-color: $main-background-color-transparent;
  border: 2px solid $main-accent-color;
  color: $main-text-color;
  font-family: $small-text-font;
  height: 45%;
  margin: {
    left: auto;
    right: auto;
    top: 15%;
  }
  text-align: center;
  width: 35%;
}

h3 {
  font: {
    family: $large-text-font;
    size: 2rem;
  }
}

.input-wrapper {
  display: flex;
  flex-direction: column;
  margin: {
    left: auto;
    right: auto;
  }
  text-align: left;
  width: 50%;
}

input {
  height: 30px;
  margin-bottom: 20px;
}

label {
  font-size: 0.8rem;
}

.sign-in {
  background-color: $main-accent-color;
  color: inherit;
  font: {
    size: 1rem;
    weight: bold;
  }
  height: 30px;
  margin: {
    bottom: 10px;
    left: auto;
    right: auto;
  }
  outline: none;
  width: 50%;
  &:active {
    background-color: black;
  }
}

#back-home {
  color: $main-text-color;
  font-size: 0.8rem;
  margin: {
    left: auto;
    right: auto;
  }
}
