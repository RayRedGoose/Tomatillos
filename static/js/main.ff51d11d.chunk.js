(this["webpackJsonpreact-starter"]=this["webpackJsonpreact-starter"]||[]).push([[0],{29:function(e,t,a){e.exports=a.p+"static/media/delete.4236a780.svg"},31:function(e,t,a){e.exports=a.p+"static/media/camera.70b57f12.svg"},32:function(e,t,a){e.exports=a.p+"static/media/film-reel.d16afa6f.svg"},33:function(e,t,a){e.exports=a.p+"static/media/cancel.1d087161.svg"},35:function(e,t,a){e.exports=a(56)},40:function(e,t,a){},46:function(e,t,a){},47:function(e,t,a){},48:function(e,t,a){},50:function(e,t,a){},51:function(e,t,a){},52:function(e,t,a){},53:function(e,t,a){},54:function(e,t,a){},55:function(e,t,a){},56:function(e,t,a){"use strict";a.r(t);var n=a(0),r=a.n(n),i=a(19),c=a.n(i),s=(a(40),a(3)),o=a(4),l=a(28),u=a(34),d=Object(o.combineReducers)({movies:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:[],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"ADD_MOVIES":return t.movies;default:return e}},ratings:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:[],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"ADD_RATINGS":return t.ratings;case"ADD_RATING":return[].concat(Object(u.a)(e),[t.rating]);default:return e}},isLoaded:function(){var e=arguments.length>0&&void 0!==arguments[0]&&arguments[0],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"IS_LOADED":return t.isLoaded;default:return e}},error:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:"",t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"HAS_ERROR":return t.error;default:return e}},user:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:null,t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"ADD_USER":return t.user;case"REMOVE_USER":return null;default:return e}},selectedMovie:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:null,t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"ADD_SELECTED_MOVIE":return t.id;default:return e}}}),m=a(2),p=a.n(m),h=a(5),f=a(6),v=a(8),g=a(7),b=a(9),E=a(11),w=(a(46),a(47),a(48),a(17)),O=a(29),k=a.n(O),j=function(e){function t(e){var a;return Object(h.a)(this,t),(a=Object(v.a)(this,Object(g.a)(t).call(this,e))).redirect=function(){a.setState({isRedirected:!0})},a.state={isRedirected:!1},a}return Object(b.a)(t,e),Object(f.a)(t,[{key:"render",value:function(){var e=this.props,t=e.id,a=e.img,n=e.title,i=e.rate,c=e.deleteButton,s={backgroundImage:'url("'.concat(a,'")'),backgroundSize:"100%",backgroundPosition:"center",width:"230px",height:"300px"};return this.state.isRedirected?r.a.createElement(w.a,{to:"/movies/".concat(t)}):r.a.createElement("section",{className:"small-movie-card"},r.a.createElement("header",{style:s},c&&r.a.createElement("img",{src:k.a,alt:"delete icon"})),r.a.createElement("p",{className:"rate-number"},i),r.a.createElement("div",{className:"title-block"},r.a.createElement("p",{className:"title",onClick:this.redirect},n)))}}]),t}(n.Component),y=(a(50),a(31)),x=a.n(y),C=a(32),S=a.n(C),D=function(){return r.a.createElement("div",{className:"loading"},r.a.createElement("img",{src:x.a,alt:"camera"}),r.a.createElement("div",{className:"light"}),r.a.createElement("img",{src:S.a,alt:"film",className:"movie-circle"}))},R=function(e){function t(){var e,a;Object(h.a)(this,t);for(var n=arguments.length,i=new Array(n),c=0;c<n;c++)i[c]=arguments[c];return(a=Object(v.a)(this,(e=Object(g.a)(t)).call.apply(e,[this].concat(i)))).createCards=function(e){return e.map((function(e){return r.a.createElement(j,{key:e.id,id:e.id,img:e.poster_path,rate:e.average_rating.toFixed(1),title:e.title})}))},a}return Object(b.a)(t,e),Object(f.a)(t,[{key:"render",value:function(){var e=this.props,t=e.movies,a=e.isLoaded,n=e.error,i=""!==n?r.a.createElement("h2",{className:"error"},n):r.a.createElement(D,null);return r.a.createElement("main",{className:"movies-container"},a&&""===n?this.createCards(t):i)}}]),t}(n.Component),N=Object(s.b)((function(e){return{movies:e.movies,isLoaded:e.isLoaded,error:e.error}}))(R),_=(a(51),function(e){return{type:"ADD_MOVIES",movies:e}}),A=function(e){return{type:"IS_LOADED",isLoaded:e}},M=function(e){return{type:"HAS_ERROR",error:e}},L=function(e){return{type:"ADD_RATINGS",ratings:e}},I=function(e){return{type:"ADD_RATING",rating:e}},T=function(e){return{type:"ADD_USER",user:e}},U=function(){return{type:"REMOVE_USER",user:null}},P=function(e){return{type:"ADD_SELECTED_MOVIE",id:e}},F=Object(s.b)((function(e){return{user:e.user}}),(function(e){return Object(o.bindActionCreators)({removeUser:U},e)}))((function(e){var t,a=e.user,n=e.removeUser;return t=a?r.a.createElement("button",{id:"logout",className:"toggleLogin",onClick:n},"Logout"):r.a.createElement(E.b,{id:"login-link",to:"/login"},r.a.createElement("button",{id:"login",className:"toggleLogin"},"Login")),r.a.createElement("header",{className:"header"},r.a.createElement("h1",null,"Rancid Tomatillos"),t)})),V=(a(52),function(e){function t(){var e,a;Object(h.a)(this,t);for(var n=arguments.length,i=new Array(n),c=0;c<n;c++)i[c]=arguments[c];return(a=Object(v.a)(this,(e=Object(g.a)(t)).call.apply(e,[this].concat(i)))).createCards=function(e,t){return t.map((function(t){var a=e.find((function(e){return e.id===t.movie_id}));return r.a.createElement(j,{key:t.id,id:a.id,img:a.poster_path,rate:t.rating,title:a.title,deleteButton:!0})}))},a}return Object(b.a)(t,e),Object(f.a)(t,[{key:"render",value:function(){var e=this.props,t=e.movies,a=e.ratings,n=e.user,i=n?"You don't have any rating yet!":"Please, log in to see your ratings!";return r.a.createElement("div",{className:"ratings"},r.a.createElement(E.c,{to:"/"},"\u2190 back to all movies"),n&&0!==a.length?this.createCards(t,a):r.a.createElement("h2",{className:"error"},i))}}]),t}(n.Component)),G=Object(s.b)((function(e){return{movies:e.movies,ratings:e.ratings,user:e.user}}))(V),B=(a(53),a(54),a(33)),J=a.n(B),W=function(){var e,t;return p.a.async((function(a){for(;;)switch(a.prev=a.next){case 0:return"https://rancid-tomatillos.herokuapp.com/api/v1/movies",a.next=3,p.a.awrap(fetch("https://rancid-tomatillos.herokuapp.com/api/v1/movies"));case 3:if((e=a.sent).ok){a.next=6;break}throw new Error("There was an error getting movies.");case 6:return a.next=8,p.a.awrap(e.json());case 8:return t=a.sent,a.abrupt("return",t);case 10:case"end":return a.stop()}}))},Y=function(e){var t,a,n;return p.a.async((function(r){for(;;)switch(r.prev=r.next){case 0:return t="https://rancid-tomatillos.herokuapp.com/api/v1/users/".concat(e,"/ratings"),r.next=3,p.a.awrap(fetch(t));case 3:if((a=r.sent).ok){r.next=6;break}throw new Error("There was an error getting ratings.");case 6:return r.next=8,p.a.awrap(a.json());case 8:return n=r.sent,r.abrupt("return",n);case 10:case"end":return r.stop()}}))},z=function(e,t){var a,n,r,i;return p.a.async((function(c){for(;;)switch(c.prev=c.next){case 0:return a="https://rancid-tomatillos.herokuapp.com/api/v1/users/".concat(t,"/ratings"),n={method:"POST",body:JSON.stringify(e),headers:{"Content-Type":"application/json"}},c.next=4,p.a.awrap(fetch(a,n));case 4:if((r=c.sent).ok){c.next=7;break}throw new Error("There was an error posting rating.");case 7:return c.next=9,p.a.awrap(r.json());case 9:return i=c.sent,c.abrupt("return",i);case 11:case"end":return c.stop()}}))},H=function(e){var t,a,n;return p.a.async((function(r){for(;;)switch(r.prev=r.next){case 0:return"https://rancid-tomatillos.herokuapp.com/api/v1/login",t={method:"POST",body:JSON.stringify(e),headers:{"Content-Type":"application/json"}},r.next=4,p.a.awrap(fetch("https://rancid-tomatillos.herokuapp.com/api/v1/login",t));case 4:if((a=r.sent).ok){r.next=7;break}throw new Error("There was a problem signing in");case 7:return r.next=9,p.a.awrap(a.json());case 9:return n=r.sent,r.abrupt("return",n);case 11:case"end":return r.stop()}}))},$=function(e){function t(e){var a;return Object(h.a)(this,t),(a=Object(v.a)(this,Object(g.a)(t).call(this,e))).toggleClicked=function(){a.setState({isClicked:!a.state.isClicked})},a.chooseRating=function(e){var t,n,r,i,c;return p.a.async((function(s){for(;;)switch(s.prev=s.next){case 0:return t=a.props,n=t.movie_id,r=t.addRating,i=parseInt(e.target.getAttribute("id").replace("number","")),c={movie_id:n,rating:i},s.next=5,p.a.awrap(z(c,a.props.user_id));case 5:r(c);case 6:case"end":return s.stop()}}))},a.state={isClicked:!1},a}return Object(b.a)(t,e),Object(f.a)(t,[{key:"render",value:function(){var e=this,t=new Array(10).fill(void 0).map((function(t,a){return r.a.createElement("p",{key:"number".concat(a+1),className:"rate-number",onClick:e.chooseRating,id:"number".concat(a+1)},a+1)}));return this.state.isClicked?r.a.createElement("section",{className:"rating-box"},r.a.createElement("h3",null,"Choose rating"),r.a.createElement("div",null,t,r.a.createElement("img",{onClick:this.toggleClicked,src:J.a,alt:"cancel"}))):r.a.createElement("button",{className:"add-rating",onClick:this.toggleClicked},"add rating")}}]),t}(n.Component),q=Object(s.b)(null,(function(e){return Object(o.bindActionCreators)({addRating:I},e)}))($),K=function(e){function t(e){var a;return Object(h.a)(this,t),(a=Object(v.a)(this,Object(g.a)(t).call(this,e))).redirect=function(){a.setState({isRedirected:!0})},a.state={isRedirected:!1},a}return Object(b.a)(t,e),Object(f.a)(t,[{key:"render",value:function(){var e=this.props,t=e.movie,a=e.rating,n=e.user,i=t.id,c=t.title,s=t.release_date,o=t.poster_path,l=t.overview,u=t.average_rating,d=s.split("-")[0],m={backgroundImage:"url(".concat(o,")"),backgroundSize:"100%",backgroundPosition:"center",width:"230px",height:"300px"};return this.state.isRedirected?r.a.createElement(w.a,{to:"/"}):r.a.createElement("main",{className:"single-movie-info"},r.a.createElement("button",{onClick:this.redirect},"\u2190 back to all movies"),r.a.createElement("div",{className:"title-div"},r.a.createElement("h2",null,"".concat(c," (").concat(d,")"))),r.a.createElement("div",{className:"img-container",style:m}),r.a.createElement("section",{className:"overview-box"},r.a.createElement("h3",null,"Avr.Rating: ",u.toFixed(1)),n&&a&&r.a.createElement("h3",null,"Your rating: ".concat(a.rating)),r.a.createElement("p",null,l),r.a.createElement("footer",null,n&&!a&&r.a.createElement(q,{movie_id:i,user_id:n.id}))))}}]),t}(n.Component),Q=Object(s.b)((function(e){var t=e.movies,a=e.ratings,n=e.user,r=e.selectedMovie;return{movie:t.find((function(e){return e.id===parseInt(r)})),rating:a.find((function(e){return e.movie_id===parseInt(r)})),user:n}}))(K),X=(a(55),function(e){function t(e){var a;return Object(h.a)(this,t),(a=Object(v.a)(this,Object(g.a)(t).call(this,e))).handleChange=function(e){var t=e.target;switch(a.state.email&&a.state.password&&a.setState({errorMsg:""}),t.id){case"email":a.setState({email:t.value});break;case"password":a.setState({password:t.value});break;default:console.log("hits default")}},a.validateForm=function(){return a.state.email&&a.state.password?(a.setState({errorMsg:""}),!0):(a.setState({errorMsg:"Please enter an email and a password"}),!1)},a.handleSubmit=function(e){if(e.preventDefault(),a.validateForm()){var t={email:a.state.email,password:a.state.password};a.fetchUser(t)}},a.fetchUser=function(e){var t;return p.a.async((function(n){for(;;)switch(n.prev=n.next){case 0:return n.prev=0,n.next=3,p.a.awrap(H(e));case 3:t=n.sent,a.props.addUser(t.user),a.props.history.push("/ratings"),a.setState({email:"",password:""}),n.next=12;break;case 9:n.prev=9,n.t0=n.catch(0),a.setState({errorMsg:"Your email or password was incorrect"});case 12:case"end":return n.stop()}}),null,null,[[0,9]])},a.state={email:"",password:"",errorMsg:""},a}return Object(b.a)(t,e),Object(f.a)(t,[{key:"render",value:function(){var e=this;return r.a.createElement("form",{className:"login-form"},r.a.createElement("h3",null,"Please sign in to continue"),r.a.createElement("div",{className:"input-wrapper"},r.a.createElement("div",{className:"error"},this.state.errorMsg),r.a.createElement("label",null,"Email:"),r.a.createElement("input",{autoComplete:"off",id:"email",value:this.state.email,type:"text",onChange:function(t){return e.handleChange(t)}}),r.a.createElement("label",null,"Password:"),r.a.createElement("input",{id:"password",value:this.state.password,type:"password",onChange:function(t){return e.handleChange(t)}}),r.a.createElement("button",{className:"sign-in",onClick:function(t){return e.handleSubmit(t)}},"Sign in"),r.a.createElement(E.b,{id:"back-home",to:"/"},"Go back")))}}]),t}(n.Component)),Z=Object(w.f)(Object(s.b)(null,(function(e){return Object(o.bindActionCreators)({addUser:T},e)}))(X)),ee=function(e){function t(){var e,a;Object(h.a)(this,t);for(var n=arguments.length,r=new Array(n),i=0;i<n;i++)r[i]=arguments[i];return(a=Object(v.a)(this,(e=Object(g.a)(t)).call.apply(e,[this].concat(r)))).fetchMoviesData=function(){var e,t,n,r,i;return p.a.async((function(c){for(;;)switch(c.prev=c.next){case 0:return e=a.props,t=e.addMovies,n=e.addLoaded,r=e.hasError,c.prev=1,c.next=4,p.a.awrap(W());case 4:i=c.sent,t(i.movies),n(!0),c.next=13;break;case 9:c.prev=9,c.t0=c.catch(1),n(!1),r(c.t0.message);case 13:case"end":return c.stop()}}),null,null,[[1,9]])},a.fetchRatingsData=function(){var e,t,n,r;return p.a.async((function(i){for(;;)switch(i.prev=i.next){case 0:return e=a.props,t=e.addRatings,n=e.user,i.prev=1,i.next=4,p.a.awrap(Y(n.id));case 4:r=i.sent,t(r.ratings),i.next=11;break;case 8:i.prev=8,i.t0=i.catch(1),t([]);case 11:case"end":return i.stop()}}),null,null,[[1,8]])},a}return Object(b.a)(t,e),Object(f.a)(t,[{key:"componentDidMount",value:function(){this.props.isLoaded||this.fetchMoviesData(),this.props.user&&this.fetchRatingsData()}},{key:"render",value:function(){var e=this;return r.a.createElement("div",{className:"App"},r.a.createElement(w.b,{exact:!0,path:"/",render:function(){return e.props.user&&e.fetchRatingsData(),r.a.createElement(r.a.Fragment,null,r.a.createElement(F,null),e.props.user&&r.a.createElement(E.c,{to:"/ratings"},"\u2605 See your ratings"),r.a.createElement(N,null))}}),r.a.createElement(w.b,{path:"/ratings",render:function(){return e.props.user&&e.fetchRatingsData(),r.a.createElement(r.a.Fragment,null,r.a.createElement(F,null),r.a.createElement(G,null))}}),r.a.createElement(w.b,{path:"/login",render:function(){return r.a.createElement(Z,null)}}),r.a.createElement(w.b,{path:"/movies/:id",render:function(t){var a=t.match;return e.props.addSelectedMovie(a.params.id),r.a.createElement(r.a.Fragment,null,r.a.createElement(F,null),r.a.createElement(Q,null))}}))}}]),t}(n.Component),te=Object(s.b)((function(e){var t=e.isLoaded;return{user:e.user,isLoaded:t}}),(function(e){return Object(o.bindActionCreators)({addMovies:_,addLoaded:A,hasError:M,addRatings:L,addSelectedMovie:P},e)}))(ee);Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));var ae=Object(o.createStore)(d,Object(l.composeWithDevTools)()),ne=r.a.createElement(s.a,{store:ae},r.a.createElement(E.a,null,r.a.createElement(te,null)));c.a.render(ne,document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()}))}},[[35,1,2]]]);
//# sourceMappingURL=main.ff51d11d.chunk.js.map