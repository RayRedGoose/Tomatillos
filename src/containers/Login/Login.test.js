import React from 'react';
import { Login, mapDispatchToProps } from './Login.js';
import { shallow } from 'enzyme';

describe('Login', () => {
  let container, instance, mockEvent;

  beforeEach(() => {
    container = shallow(<Login addUser={jest.fn()}/>);
    instance = container.instance();
    mockEvent = {
      target: {
        id: 'email',
        value: 'new value',
      },
      preventDefault: jest.fn()
    }
  });

  it('Should match the snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('Should attempt to login when the button is clicked', () => {
    instance.handleSubmit = jest.fn();
    container.find('.sign-in').simulate('click');
    expect(instance.handleSubmit).toHaveBeenCalled();
  });

  it('Should have blank strings as a default for all state values', () => {
    expect(instance.state).toEqual({
      email: '',
      password: '',
      errorMsg: ''
    });
  });

  describe('validateForm', () => {

    it('Should update the error message if there is no email and password', () => {
      instance.validateForm();
      expect(instance.state.errorMsg).toEqual('Please enter an email and a password');
    });

    it('Should return false if there is no email and password entered', () => {
      expect(instance.validateForm()).toEqual(false);
    });

    it('Should return true if there is an email and password entered', () => {
      instance.state = {
        email: 'test email',
        password: 'test password',
        errorMsg: ''
      }
      expect(instance.validateForm()).toEqual(true);
    });

  });

  describe('handleChange', () => {

    it('Should remove the error message if the fields are filled in', () => {
      instance.state = {
        email: 'test email',
        password: 'test password',
        errorMsg: 'test error'
      }
      instance.handleChange(mockEvent);
      expect(instance.state.errorMsg).toEqual('');
    });

    it('Should update the email in state if the target is the email input', () => {
      instance.handleChange(mockEvent);
      expect(instance.state.email).toEqual('new value');
    });

    it('Should update the password in state if the target is the password input', () => {
      mockEvent = {
        target: {
          id: 'password',
          value: 'new value'
        }
      }
      instance.handleChange(mockEvent);
      expect(instance.state.password).toEqual('new value');
    });

  });

  describe('handleSubmit', () => {

    it('Should update the error message in state if the form isn\'t validated', () => {
      instance.handleSubmit(mockEvent);
      expect(instance.state).toEqual({
        email: '',
        password: '',
        errorMsg: 'Please enter an email and a password'
      });
    });

    it('Should clear the state if the form is validated',  () => {
      instance.state = {
        email: 'test email',
        password: 'test password',
        errorMsg: ''
      };
      instance.handleSubmit(mockEvent);
      expect(instance.state).toEqual({
        email: '',
        password: '',
        errorMsg: ''
      });
    });

  });

});
