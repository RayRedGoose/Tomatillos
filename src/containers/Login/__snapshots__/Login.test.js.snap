// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Should match the snapshot 1`] = `
<form
  className="login-form"
>
  <h3>
    Please sign in to continue
  </h3>
  <div
    className="input-wrapper"
  >
    <div
      className="error"
    />
    <label>
      Email:
    </label>
    <input
      autoComplete="off"
      id="email"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label>
      Password:
    </label>
    <input
      id="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <button
      className="sign-in"
      onClick={[Function]}
    >
      Sign in
    </button>
    <Link
      id="back-home"
      to="/"
    >
      Go back
    </Link>
  </div>
</form>
`;
